# Original Data clean up

In [2]:
#Imports
import pandas as pd
import numpy as np


In [3]:
port = pd.read_csv("../data/raw/Port_of_Los_Angeles_-_Historical_TEU_Statistics_20260527.csv")
env = pd.read_csv("../data/raw/calenviroscreen50csv_d_12226.csv")

## Port of Los Angeles TEU Data CleanUp

In [19]:
port.head()

,Year,TEUs in Million
0,1980,0.633
1,1981,0.476
2,1982,0.606
3,1983,0.734
4,1984,0.908


## CalEnviroScreen Data Clean Up

In [7]:
env.head()
#port.head()
print(env.columns.tolist())

['tract', 'ZIP', 'apx_loc', 'county', 'TotalPop', 'CIscore', 'CIscoreP', 'ozone', 'ozoneP', 'pm', 'pmP', 'lead', 'leadP', 'diesel', 'dieselP', 'drink', 'drinkP', 'pest', 'pestP', 'RSEIhaz', 'RSEIhazP', 'traffic', 'trafficP', 'cleanups', 'cleanupsP', 'gwthreats', 'gwthreatsP', 'haz', 'hazP', 'iwb', 'iwbP', 'SmATS', 'SmATSP', 'swis', 'swisP', 'Pollution', 'PollutionScore', 'PollutionP', 'asthma', 'asthmaP', 'cvd', 'cvdP', 'diabetes', 'diabetesP', 'lbw', 'lbwP', 'edu', 'eduP', 'housingB', 'housingBP', 'ling', 'lingP', 'pov', 'povP', 'unemp', 'unempP', 'PopChar', 'PopCharScore', 'PopCharP', 'popund_10', 'pop10_64', 'popovr_64', 'white_pct', 'hispanic_pct', 'black_pct', 'nativeamer_pct', 'asian_pct', 'othmulti_pct']


### The original CalEnviroScreen dataset contains 68 variables related to pollution exposure, environmental conditions, health outcomes, and demographic characteristics. While additional variables such as poverty, housing burden, education, and demographic composition may be useful for exploring environmental justice impacts, the initial analysis focuses on variables most relevant to the project's objective. Pollution Score, PM2.5, Diesel PM, Traffic, Ozone, Asthma, Cardiovascular Disease, Diabetes, and Low Birth Weight were selected because they capture both environmental exposures and potential health effects associated with port activity while keeping the dataset focused and manageable.


In [15]:
env_clean = env[
    [
        "tract",
        "county",
        "apx_loc",
        "TotalPop",
        "PollutionScore",
        "pm",
        "diesel",
        "traffic",
        "ozone",
        "asthma",
        "cvd",
        "diabetes",
        "lbw"
    ]
]
env_clean.head()
env_clean = env_clean.rename(columns={
    "tract": "census_tract",
    "apx_loc": "location",
    "TotalPop": "population",
    "pm": "pm25",
    "diesel": "diesel_pm",
    "traffic": "traffic_density",
    "cvd": "cardiovascular_disease",
    "lbw": "low_birth_weight"
})
env_clean.head()

,census_tract,county,location,population,PollutionScore,pm25,diesel_pm,traffic_density,ozone,asthma,cardiovascular_disease,diabetes,low_birth_weight
0,6001400100,Alameda,Berkeley,3094,4.369906,7.135387,0.095739,934.715,0.029251,17.36,7.29,9.1,7.03
1,6001400200,Alameda,Oakland,2093,5.046525,7.348713,0.171673,1753.647,0.028516,20.69,7.04,7.0,4.13
2,6001400300,Alameda,Oakland,5727,4.971513,7.446539,0.155387,1617.707,0.028516,26.72,7.51,7.4,4.51
3,6001400400,Alameda,Oakland,4395,4.986183,7.451338,0.160767,1253.706,0.028516,37.93,8.89,6.5,3.06
4,6001400500,Alameda,Oakland,3822,4.496636,7.451338,0.133439,484.944,0.028516,49.55,9.87,7.2,5.50


In [23]:
env_clean.isnull().sum()
env_clean = env_clean.dropna()
env_clean.shape
env_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8859 entries, 0 to 9104
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   census_tract            8859 non-null   int64  
 1   county                  8859 non-null   object 
 2   location                8859 non-null   object 
 3   population              8859 non-null   int64  
 4   PollutionScore          8859 non-null   float64
 5   pm25                    8859 non-null   float64
 6   diesel_pm               8859 non-null   float64
 7   traffic_density         8859 non-null   float64
 8   ozone                   8859 non-null   float64
 9   asthma                  8859 non-null   float64
 10  cardiovascular_disease  8859 non-null   float64
 11  diabetes                8859 non-null   float64
 12  low_birth_weight        8859 non-null   float64
dtypes: float64(9), int64(2), object(2)
memory usage: 969.0+ KB


#### The CalEnviroScreen dataset was cleaned by selecting variables most relevant to the project's objective of examining the relationship between port activity, pollution exposure, and health outcomes. Variables related to overall pollution burden, specific pollutants, traffic density, and health indicators were retained, while unrelated variables were removed to improve readability and efficiency. Column names were renamed to more descriptive labels, and rows containing missing values were removed. The final cleaned dataset contains 8,859 census tracts and 13 variables, providing a consistent dataset for future analysis.


In [25]:
env_clean.to_csv(
    "../data/cleaned/env_clean.csv",
    index=False
)
port.to_csv(
    "../data/cleaned/port_clean.csv",
    index=False
)